930-delegate-tasks/src/main/java/io/harness/delegate/utils/ServiceHookHandler.java

```
@@ -0,0 +1,167 @@
+  }
+
+  public void addContext(String key, String value) {
+    context.put(key, value);
+  }
+
+  public void execute(ServiceHookType type, ServiceHookAction action, String workingDirectory, LogCallback logCallback,
+      String contextDirectory) {
+    switch (action) {
+      case FETCH_FILES:
+        addContext("MANIFEST_FILES_DIRECTORY", contextDirectory);
+        break;
+      case TEMPLATE_MANIFEST:
+        addContext("MANIFEST_FILE_OUTPUT_PATH", contextDirectory);
+        break;
+      case STEADY_STATE_CHECK:
+        addContext("WORKLOADS_LIST", contextDirectory);
+        break;
+      default:
+        break;
+    }
+    Map<String, String> hooksToApply = requiredHooks(action.getActionName(), type.getName());
+    if (isEmpty(hooksToApply)) {
+      return;
+    }
+    hooksToApply.forEach((k, v) -> {
+      String directory = Paths.get(workingDirectory, ".__harness_internal_hooks", k).toString();
+      logCallback.saveExecutionLog(format("Starting %s ", type.getName()));
+      ProcessResult processResult =
+          executeCommand(context, v, directory, format(errorMessage, type.getName(), action.getActionName()));
+      logCallback.saveExecutionLog(format("Hook output %s", processResult.getOutput().getUTF8()));
+    });
+    return;
+  }
+
+  private ProcessResult executeCommand(
+      Map<String, String> envVars, String command, String directoryPath, String errorMessage) {
+    ProcessExecutor processExecutor = createProcessExecutor(command, directoryPath, envVars);
+
+    return executeCommand(processExecutor, errorMessage);
+  }
+
+  private ProcessResult executeCommand(ProcessExecutor processExecutor, String errorMessage) {
+    errorMessage = isEmpty(errorMessage) ? "" : errorMessage;
+    try {
+      createDirectoryIfDoesNotExist(Paths.get(String.valueOf(processExecutor.getDirectory())));
+      return processExecutor.execute();
```

Add handle for process result output code is not 0, it should throw an error